relayer/pkg/chainlink/txm/txstore.go

```diff
@@ -57,8 +66,11 @@ func (s *TxStore) Confirm(hash string) error {
 	defer s.lock.Unlock()
 
 	if nonce, exists := s.hashToNonce[hash]; exists {
-		delete(s.hashToNonce, hash)
 		delete(s.nonceToHash, nonce)
+
```

Comment:
correct - ordering it would help or could also assume that if a higher nonce is confirmed, lower nonces must also have been confirmed (like how EVM does it - `confirmed without receipt`)

Reply:
I haven't seen this happen in the logs of the soak test. tracking the highest nonce seems to be easier. I can do that

Reply:
Actually, I'm not following. So is the issue is that nonce=2 could be confirmed before nonce=1 is confirmed? That can definitely happen, but that wouldn't break any logic of Save() because that higher nonce had to have been Saved() first in the broadcast so even though the nonce is gone from the store the store's currentNonce has to be higher than the confirmed tx

Reply:
**t=0 (add two txs)**

tx_store:
{nonce-1, nonce-2}

currNonce:
3

**t=1 (later tx is confirmed first)**

tx_store:
{nonce-1}

currNonce:
3

**t=2 (broadcast new tx)**

tx_store:
{nonce-1, nonce-3}

currNonce:
4

Reply:
oh i think i see what you mean, the Save() logic doesn't save currentNonce based off of the nonce it was given

Reply:
it doesn't start counting off at nonce=1. It starts counting off at the highest unconfirmed nonce ever saved in the txm. So the fact that a nonce was confirmed doesn't remove the tx store's memory that currentNonce is higher than any of the unconfirmed nonces seen